server/src/main/java/org/elasticsearch/transport/RemoteClusterService.java

```diff
@@ -667,6 +680,13 @@ public RemoteClusterClient getRemoteClusterClient(
                 "this node does not have the " + DiscoveryNodeRole.REMOTE_CLUSTER_CLIENT_ROLE.roleName() + " role"
             );
         }
+        if (isStateless && disconnectedStrategy == DisconnectedStrategy.RECONNECT_UNLESS_SKIP_UNAVAILABLE) {
+            throw new IllegalArgumentException(
+                "DisconnectedStrategy ["
+                    + DisconnectedStrategy.RECONNECT_UNLESS_SKIP_UNAVAILABLE
+                    + "] is not supported in stateless environments"
+            );
+        }
```

Comment:
Does it make sense to also `assert false : "message"` here? It must be a programming error (existing or future) if it happens?